client/src/util/exponentialBackoffFetch.js

```diff
@@ -28,29 +28,43 @@ export default function createBackoffFetch(retryTimes = 3) {
       delayFetch();
 
       function delayFetch() {
-        timeoutId = setTimeout(async () => {
-          try {
-            const response = await fetch(...fetchArgs);
-
-            if (!response.ok) {
-              const { status, statusText } = response;
-
-              throw Error(`${status}: ${statusText}`);
-            }
-
-            resolve(response);
-          } catch (error) {
-            if (retryCount === retryTimes) {
-              reject(error);
-            }
-
-            retryCount += 1;
+        if (retryCount === 0) {
+          fetchAndHandle();
+          // delay amount (3.5, 17.5, 87.5, 437.5 seconds)
+        } else timeoutId = setTimeout(fetchAndHandle, 700 * 5 ** retryCount);
+      }
+      async function fetchAndHandle() {
+        try {
+          promise.response = await mockFetch(...fetchArgs);
 
-            delayFetch();
+          if (promise.response.ok || retryCount === retryTimes) {
+            resolve(promise.response);
           }
-          // delay amount (3.5, 17.5, 87.5, 437.5 seconds)
-        }, retryCount * 700 * 5 ** retryCount);
+          console.log("success", fetchArgs[2]);
+        } catch (error) {
+          console.log("error", fetchArgs[2]);
+          if (retryCount === retryTimes) {
+            reject(error);
+          }
```

Comment:
We're only going to serve fetch API error if it's the one that pushes us over the retry amounts.  (Maybe we serve a list of fall the errors instead?)